src/SDDK/memory.hpp

```diff
@@ -657,18 +658,18 @@ void memory_pool_deleter::memory_pool_deleter_impl::free(void* ptr__)
 #ifdef NDEBUG
 #define mdarray_assert(condition__)
 #else
-#define mdarray_assert(condition__)                                  \
-{                                                                    \
-    if (!(condition__)) {                                            \
-        std::printf("Assertion (%s) failed ", #condition__);         \
-        std::printf("at line %i of file %s\n", __LINE__, __FILE__);  \
-        std::printf("array label: %s\n", label_.c_str());            \
-        for (int i = 0; i < N; i++) {                                \
-            std::printf("dim[%i].size = %li\n", i, dims_[i].size()); \
-        }                                                            \
-        raise(SIGABRT);                                              \
-    }                                                                \
-}
+#define mdarray_assert(condition__)                                                                                    \
+    {                                                                                                                  \
+        if (!(condition__)) {                                                                                          \
+            std::fprintf(stderr, "Assertion (%s) failed ", #condition__);                                              \
+            std::fprintf(stderr, "at line %i of file %s\n", __LINE__, __FILE__);                                       \
+            std::fprintf(stderr, "array label: %s\n", label_.c_str());                                                 \
+            for (int i = 0; i < N; i++) {                                                                              \
+                std::fprintf(stderr, "dim[%i].size = %li\n", i, dims_[i].size());                                      \
+            }                                                                                                          \
+            MPI_Abort(MPI_COMM_WORLD, 1);                                                                              \
```

Comment:
I want to object to this change. First, memory.hpp (memory management functions) now depends on MPI header. Why should they? Second, with raise(SIGABRT) I can get a core dump. This is super valuable on the parallel debugs on Daint. With MPI_Abort the only alternative would be to use distributed debugger, probably with the GUI. Otherwise how I would trace the source of assertion?

Reply:
Good point about core dumps, they used to be uploaded to Gitlab before we had `ctest` to run tests. Would be neat to bring that feature back to debug CI segfaults :). Also I'm thinking maybe we had `--unbuffered` in slurm before to ensure data would actually print before a segfault, is that correct @toxa81? Or how else can we explain there's no assert statement being printed?

Reply:
Yes, some good points. The hope was that `MPI_Abort` solves the deadlock issue, if an assertion is triggered in CI. Turns out, it doesn't actually help. I'll revert that change.

Reply:
yes, ` --unbuffered` tells slurm to write output immediately instead of storing it in some buffer and writing it in chunks